src/Microsoft.AspNetCore.Mvc.Razor/RazorPage.cs

```diff
@@ -800,7 +800,8 @@ public Task<HtmlString> RenderSectionAsync(string name, bool required)
             else if (required)
             {
                 // If the section is not found, and it is not optional, throw an error.
-                throw new InvalidOperationException(Resources.FormatSectionNotDefined(sectionName, Path));
+                var message = Resources.FormatSectionNotDefined(sectionName, ViewContext.View.Path);
```

Comment:
Someone more familiar with the internals of view/viewcontext might want to double check my logic that ViewContext.View.Path is always what we're looking for here.